pkg/kubelet/kubelet.go

```diff
@@ -1002,6 +1002,10 @@ type Kubelet struct {
 	cloudproviderRequestSync chan int
 	// Request timeout
 	cloudproviderRequestTimeout time.Duration
+	// Last list of node addresses retrieved from the cloud provider
+	cloudproviderLastNodeAddresses []v1.NodeAddress
+	// Last error retrieved from the cloud provider
+	cloudproviderLastError error
```

Comment:
why is this needed here as member var and not just in the kubelet_node_status code block?

Reply:
Both variables needs to be on the global level so the go routine does not store its values into a local variable that disappears. Plus, the cloud related code is going out at some point and the variables are not exported. So it's fine to keep it that way. We can change it and polish later.